posthog/tasks/email.py

```diff
@@ -502,3 +502,235 @@ def send_error_tracking_issue_assigned(assignment: ErrorTrackingIssueAssignment,
+            "end_time": yesterday_end,
+        },
+    )
+
+    if not failed_functions:
+        logger.info("No HogFunctions with failures found")
+        return
+
+    # Group by team_id
+    teams_with_failures: dict[int, list[str]] = {}
+    for team_id, hog_function_id in failed_functions:
+        if team_id not in teams_with_failures:
+            teams_with_failures[team_id] = []
+        teams_with_failures[team_id].append(hog_function_id)
+
+    # Filter teams based on the feature flag setting
+    allowed_team_ids = settings.HOG_FUNCTIONS_DAILY_DIGEST_TEAM_IDS
+    if allowed_team_ids:
+        # Convert string team IDs to integers for comparison
+        allowed_team_ids_int = [int(team_id) for team_id in allowed_team_ids]
+        teams_with_failures = {
+            team_id: hog_function_ids
+            for team_id, hog_function_ids in teams_with_failures.items()
+            if team_id in allowed_team_ids_int
+        }
+        logger.info(
+            f"Filtered to {len(teams_with_failures)} teams based on HOG_FUNCTIONS_DAILY_DIGEST_TEAM_IDS setting"
+        )
+
+    if not teams_with_failures:
+        logger.info("No teams in allowed list have HogFunctions with failures")
+        return
+
+    logger.info(f"Found {len(teams_with_failures)} teams with HogFunction failures")
+
+    # Fan out to team-specific tasks
+    for team_id, failed_hog_function_ids in teams_with_failures.items():
+        send_team_hog_functions_digest.delay(team_id, failed_hog_function_ids, date_str, formatted_date)
+        logger.info(f"Scheduled digest for team {team_id} with {len(failed_hog_function_ids)} failed functions")
+
+    logger.info("Completed HogFunctions daily digest task")
+
+
+@shared_task(**EMAIL_TASK_KWARGS)
+def send_team_hog_functions_digest(
+    team_id: int, failed_hog_function_ids: list[str], date_str: str, formatted_date: str
+) -> None:
+    """
+    Send daily digest email for a specific team with their failed HogFunctions.
+    """
+    from datetime import timedelta
+    from posthog.clickhouse.client import sync_execute
+    from posthog.models.hog_functions.hog_function import HogFunction
+
+    logger.info(f"Processing HogFunctions digest for team {team_id}")
+
+    # Calculate date range for yesterday's data
+    yesterday = timezone.now() - timedelta(days=1)
+    yesterday_start = yesterday.replace(hour=0, minute=0, second=0, microsecond=0)
+    yesterday_end = yesterday.replace(hour=23, minute=59, second=59, microsecond=999999)
+
+    # Get HogFunction data from PostgreSQL for the failed functions
+    hog_functions = HogFunction.objects.filter(
+        team_id=team_id, id__in=failed_hog_function_ids, enabled=True, deleted=False
+    ).values("id", "team_id", "name", "type")
+
+    if not hog_functions:
+        logger.info(f"No active HogFunctions found for team {team_id}")
+        return
+
+    # Get metrics data from ClickHouse for these specific HogFunctions
+    metrics_query = """
```

Comment:
Really feels like this could be DRYed up between the two tasks. Really we don't need to filter on the function_ids here, we just need the team_id as we know there are failures